backend/Auth/ResetPassword.ts

```diff
@@ -64,3 +68,67 @@ router.post(
     return res.status(204).end();
   }
 );
+
+router.get<Params>(
+  "/validate-token/:token",
+  param("token").exists(),
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return invalidRequest(res, errors);
+    }
+    const token = req.params.token;
+    const tokenType = req.query?.tokenType;
```

Comment:
The optional chaining here could be avoided with better types. On line 72, do `router.get(<Params, void, ResponseBody, QueryParams>)` and then define a typescript interface:

```ts
interface QueryParams {
  tokenType?: string;
}

interface ResponseBody {
  tokenIsValid: boolean;
  tokenIsExpired: boolean;
  email?: string;
}
```